jmx-scraper/src/main/resources/hbase.yaml

```diff
@@ -0,0 +1,329 @@
+---
+
+rules:
+
+  - bean: Hadoop:service=HBase,name=Master,sub=Server
+    prefix: hbase.master.
+    unit: "{server}"
+    type: updowncounter
+    mapping:
+      numDeadRegionServers:
+        metric: &metric region_server.count
+        desc: &desc The number of region servers.
+        metricAttribute:
+          state: const(dead)
+      numRegionServers:
+        metric: *metric
+        desc: *desc
+        metricAttribute:
+          state: const(live)
+
+  - bean: Hadoop:service=HBase,name=Master,sub=AssignmentManager
+    prefix: hbase.master.regions_in_transition.
+    unit: "{region}"
+    type: updowncounter
+    mapping:
+      ritCount:
+        metric: count
+        desc: The number of regions that are in transition.
+      ritCountOverThreshold:
+        metric: over_threshold
+        desc: The number of regions that have been in transition longer than a threshold time.
+      ritOldestAge:
+        metric: oldest_age
+        unit: ms
+        type: gauge
+        desc: The age of the longest region in transition.
+
+  - bean: Hadoop:service=HBase,name=RegionServer,sub=Server
+    prefix: hbase.region_server.
+    type: updowncounter
+    metricAttribute:
+      region_server: &hostname beanattr(tag\.Hostname)
+    mapping:
+      regionCount:
+        metric: region.count
+        unit: "{region}"
+        desc: The number of regions hosted by the region server.
+
```

Comment:
[minor] the spacing between blocks is intentional to keep things a bit readable here, maybe adding an explicit "header comment" with for example the metric name/prefix/suffix would help to understand this is intentional and needs to be preserved when maintaining this.

Reply:
Comments added